src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+
+void resize(const unsigned char* image_data, int s_width, int s_height,
```

Comment:
It would be great if we could take an `arma::mat` (or `arma::vec`) plus a `data::ImageInfo` here.  The `ImageInfo` holds `s_width`, `s_height`, and `num_channel`.  Then you could modify the ImageInfo and the matrix.  So the API would be:

```
template<typename eT>
void ResizeImage(Mat<eT>& images, data::ImageInfo& info, const size_t newWidth, const size_t newHeight);
```

However, I see that the STB API expects either `uint8`s, `uint16`s, or `float`s... this is a little unfortunate.  It means that we may have to convert whatever the input type is.

Once you have converted the matrix, though, you should be able to get the `image_data` argument as `m.colptr(i)` where `i` is the index of the image in the matrix.

If you created an output matrix of the right size (`newWidth * newHeight * numChannels` rows by `images.n_cols` columns), then you can use `outputMatrix.colptr(i)` as `frame_buffer_out`.

I think that should all work, let me know if I can clarify anything.